index.js

```diff
@@ -2,328 +2,416 @@
-let resourceClient, computeClient, storageClient, networkClient, keyVaultManagementClient, keyVaultClient, graphClient;
+let clientId = process.env["AZURE_CLIENT_ID"];
+let domain = process.env["AZURE_TENANT_ID"];
+let secret = process.env["AZURE_CLIENT_SECRET"];
+let objectId = process.env["AZURE_OBJECT_ID"];
+let subscriptionId = process.env["AZURE_SUBSCRIPTION_ID"];
+let resourceClient,
+  computeClient,
+  networkClient,
+  keyVaultManagementClient,
+  certificateClient,
+  graphClient;
+// let computeClient = new ComputeManagementClient(credentials, subscriptionId);
 
 //Sample Config
 let randomIds = {};
-let location = 'westus';
-let resourceGroupName = _generateRandomId('testrg', randomIds);
-let vmName = _generateRandomId('testvm', randomIds);
-let vnetName = _generateRandomId('testvnet', randomIds);
-let subnetName = _generateRandomId('testsubnet', randomIds);
-let publicIPName = _generateRandomId('testpip', randomIds);
-let networkInterfaceName = _generateRandomId('testnic', randomIds);
-let ipConfigName = _generateRandomId('testcrpip', randomIds);
-let domainNameLabel = _generateRandomId('testdomainname', randomIds);
-let keyVaultName = _generateRandomId('testkv', randomIds);
-let certificateName = _generateRandomId('testcert', randomIds);
+let location = "eastus";
+let resourceGroupName = _generateRandomId("testrg", randomIds);
+let vmName = _generateRandomId("testvm", randomIds);
+let vnetName = _generateRandomId("testvnet", randomIds);
+let subnetName = _generateRandomId("testsubnet", randomIds);
+let publicIPName = _generateRandomId("testpip", randomIds);
+let networkInterfaceName = _generateRandomId("testnic", randomIds);
+let ipConfigName = _generateRandomId("testcrpip", randomIds);
+let domainNameLabel = _generateRandomId("testdomainname", randomIds);
+let keyVaultName = _generateRandomId("testkv", randomIds);
+let certificateName = _generateRandomId("testcert", randomIds);
 
 //Refs
 let vaultObj;
 let certificateSecretObj;
 
 // Ubuntu config
-let publisher = 'Canonical';
-let offer = 'UbuntuServer';
-let sku = '16.04.0-LTS';
+let publisher = "Canonical";
+let offer = "0001-com-ubuntu-server-focal";
+let sku = "20_04-lts-gen2";
 
 // Windows config
-//let publisher = 'microsoftwindowsserver';
-//let offer = 'windowsserver';
-//let sku = '2012-r2-datacenter';
+// let publisher = "microsoftwindowsserver";
+// let offer = "windowsserver";
+// let sku = "2012-r2-datacenter";
 
-let adminUsername = 'notadmin';
-let adminPassword = 'Pa$$w0rd92';
+let adminUsername = "notadmin";
+let adminPassword = "Pa$$w0rd92";
 
 ///////////////////////////////////////////
 //     Entrypoint for sample script      //
 ///////////////////////////////////////////
 
-msRestAzure.loginWithServicePrincipalSecret(clientId, secret, domain, function (err, credentials, subscriptions) {
-  if (err) return console.log(err);
+async function main() {
+  const credentials = new DefaultAzureCredential();
```

Comment:
Please use ClientSecretCredential in the case they are using msRestAzure.loginWithServicePrincipalSecret.